django/contrib/sitemaps/views.py

```diff
@@ -71,7 +72,7 @@ def sitemap(request, sitemaps, section=None,
             raise Http404("No page '%s'" % page)
     response = TemplateResponse(request, template_name, {'urlset': urls},
                                 content_type=content_type)
-    if hasattr(site, 'latest_lastmod'):
+    if 'site' in locals() and hasattr(site, 'latest_lastmod'):
```

Comment:
I don't think `'site' in locals()` is acceptable. How about `site=None` to start and then checking if `site is not None` here? A regression test and a Trac ticket are also required.